modules/data-entry/src/main/frontend/src/questionnaire/FormEntry.jsx

```diff
@@ -57,8 +59,21 @@ export const ENTRY_TYPES = QUESTION_TYPES.concat(SECTION_TYPES).concat(INFO_TYPE
  * @returns a React component that renders the question
  */
 let displayQuestion = (questionDefinition, path, existingAnswer, key, classes, onAddedAnswerPath, sectionAnswersState, onChange, pageActive, isEdit, instanceId) => {
+  const [ doHighlight, setDoHighlight ] = useState();
+  const [ anchor, setAnchor ] = useState();
+
+  const location = useLocation();
+
+  // if autofocus is needed and specified in the url
+  useEffect(() => {
+    setAnchor(decodeURIComponent(location.hash.substr(1)))
```

Comment:
Replace deprecated function call:
```suggestion
    setAnchor(decodeURIComponent(location.hash.substring(1)))
```